Add mutate.R and setLabels.R

eoduniyi opened the pull request:
Please see issue #12 for knowing what mutate.R allows and disallows

eoduniyi committed:
Add mutate.R and setLabels.R

eoduniyi committed:
Remove hy_update_labels.R and transmute.R

eoduniyi committed:
Update testthat.R

Make sure hySpc.dplyr is being referenced

eoduniyi committed:
Remove hy_update_labels from select.R

eoduniyi committed:
Update select.R

eoduniyi committed:
Add setLabels.R

eoduniyi committed:
Add mutate.R

eoduniyi committed:
Update documentation to reference hySpc.dplyr

eoduniyi committed:
Add rlang::enquos dependency

eoduniyi committed:
Update vignette for mutate.R

eoduniyi committed:
Remove hy_update_labels.R and transmute.R

eoduniyi committed:
Import fauxCell

eoduniyi committed:
Replace `fauxCell` with `chondro`

eoduniyi committed:
Replace `fauxCell` with `chondro` in vignette

cbeleites committed:
merge devel and resolve conflicts

Merge branch 'develop' into feature/12-mutate-setLabels

# Conflicts:
#	R/hy_update_labels.R
#	R/transmute.R
#	vignettes/hySpc_dplyr.Rmd

cbeleites committed:
move transmute to its own file

cbeleites committed:
delete hy_update_labels

eoduniyi committed:
Update unittest.R

cbeleites commented:
@eoduniyi the build failed because there's a leftover old package name in your branch. 

I already found & fixed it in develop, so I merged develop into your feature branch and resolved the conflicts

eoduniyi committed:
Merge branch 'feature/12-mutate-setLabels' of https://github.com/r-hyperspec/hySpc.dplyr into feature/12-mutate-setLabels

cbeleites committed:
adapt unit test to dplyr now accepting matrices in columns.

cbeleites committed:
bump dependeny to dplyr >= 1.0.0 for matrix-in-column functionality.

Also bump our version number.

cbeleites committed:
bump dependeny to dplyr >= 1.0.0 for matrix-in-column functionality.

Also bump our version number.

cbeleites committed:
Merge branch 'feature/12-mutate-setLabels' of github.com:r-hyperspec/hySpc.dplyr into feature/12-mutate-setLabels

cbeleites committed:
fix dplyr version requirement

eoduniyi committed:
Update mutate.R with dplyr 1.0.0

since dplyr 1.0.0 got published it natively works with columns containing matrices

eoduniyi committed:
Update transmute.R with dplyr 1.0.0

since dplyr 1.0.0 got published it natively works with columns containing matrices

eoduniyi committed:
Merge branch 'develop' into feature/12-mutate-setLabels

eoduniyi committed:
Update README.md

cbeleites commented on R/setLabels.R:
```diff
@@ -0,0 +1,104 @@
+#' @param .data hyperSpec object
+#' @param res resulting data frame
+#' @export
+setLabels.select <- function(.data, res) {
```
- Should `setLabels.select()` and `setLabels.rename()` really be exported?
- since `method.class()` signifies S3 methods, would snake-case names (`setLabels_select()`) be better here?

eoduniyi replied:
Okay, will change to snake-case

cbeleites commented on R/transmute.R:
```diff
@@ -1,138 +1,56 @@
     expect_error(transmute.hyperSpec(df))
   })
 
-  # UT2
```
@eoduniyi, why did you remove the unit tests?

eoduniyi replied:
UT2 was:
```R
  # UT2
  test_that("arguments are correctly passed onto mutate/transmute", {
    df <- flu
    df <- dplyr::transmute(df@data, c, c = 0, c = 1)
    expect_equivalent(transmute.hyperSpec(flu, c, c = 0, c = 1), df)
  })
```

But making sure arguments are passed in their correct order is behaviour that is the responsibility of `dplyr`

cbeleites replied:
Sure, but we still want to check that we get the correct result. We *could* have a mistake that would return an invalid hyperSpec object, or the wrong content e.g. if our code doesn't hand over the correct result form dplyr.

eoduniyi committed:
Add "arguments are correctly passed" unit test

eoduniyi committed:
Merge branch 'develop' into feature/12-mutate-setLabels

eoduniyi committed:
Update DESCRIPTION

eoduniyi committed:
Update NAMESPACE

eoduniyi committed:
Update unit test in mutate.R

eoduniyi committed:
Update unit test in transmute.R

Having difficulty with two unit test here.

eoduniyi committed:
Update setLabels.R

Change setLabels to snake-case

eoduniyi committed:
Update select.R

Add `setLabels_select`

eoduniyi committed:
Update rename.R

Add `setLabels_rename`, clean up test, and enforce tidyverse styling

eoduniyi committed:
Update transmute.R

Add `setLabels_select`

eoduniyi committed:
Update hySpc_dplyr.Rmd

Clean up vignette
